File: sdk/src/server-api/sc/api/plugins/host/IGameListener.java
package sc.api.plugins.host;

import java.util.Map;

import sc.api.plugins.IGameState;
import sc.framework.plugins.Player;
import sc.networking.InvalidScoreDefinitionException;
import sc.shared.PlayerScore;

public interface IGameListener {
  void onGameOver(Map<Player, PlayerScore> results) throws InvalidScoreDefinitionException;

  void onStateChanged(IGameState data, boolean observersOnly);

  void onPaused(Player nextPlayer);
}

File: sdk/src/main/server-api/sc/api/plugins/host/IGameListener.kt
package sc.api.plugins.host

import sc.api.plugins.IGameState
import sc.framework.plugins.Player
import sc.shared.GameResult
import sc.shared.PlayerScore

interface IGameListener {
    fun onGameOver(result: GameResult)
    fun onStateChanged(data: IGameState, observersOnly: Boolean)
}